source/org/zfin/marker/service/MarkerService.java

```diff
@@ -170,6 +173,26 @@ public static SequenceInfo getSequenceInfoSummary(Marker marker) {
                 .collect(toList());
     }
 
+    /**
+     * Group DBLinks by db name and accession number
+     * @param link
+     * @return
+     */
+    private static ImmutablePair<String, String> getDBLinkAggregationKey(DBLink link) {
+        String foreignDBName = null;
+        String accession = null;
+
+        //probably the try catch is unnecessary, but many chained gets makes me nervous
+        try {
+            foreignDBName = link.getReferenceDatabase().getForeignDB().getDbName().name();
```

Comment:
Fixed

Reply:
Also, I am not sure what this grouping / aggregation is used for. I need some more explanation. I think you are aggregating all pubs of a given DB-accession in a single DBLink. Not sure how this relates to the problem at hand.

Reply:
The changes I made were to disaggregate some entries rather than aggregate them.  
The ticket has some context in the last few messages between Doug, Ken, and I. 
The solution that we ended up with is that this:

![Screen Shot 2022-03-03 at 12 36 07 PM](https://user-images.githubusercontent.com/90803397/156649178-b67e8a96-3024-41fc-b067-c7591517bc90.png)

Should become this:

![Screen Shot 2022-03-03 at 12 37 48 PM](https://user-images.githubusercontent.com/90803397/156649217-2e27b5a0-debd-4374-8ab4-f786aa9ec25a.png)

Prior to the changes in this pull request, those two entries in the first screenshot are aggregated because the groupingBy function groups by accession number only.  I changed it to group by the combination of accession number and database name.  This matches the desired outcome Doug and Ken described in the comments on ZFIN-7799.

Reply:
Thanks for the clarifying info. Makes sense now.

Reply:
I took into account your suggestions:
1. To add comments to <String, String> (I tweaked it so the first type is informative and the second is commented: <ForeignDB, String /*Accession*/>)
2. Investigate nesting "groupingBy" calls. I looked into this, but don't think it gives too much advantage in terms of readability or concision.  

I think you were suggesting something like what is described here:

https://www.javacodegeeks.com/2016/03/multi-level-grouping-streams.html
![image](https://user-images.githubusercontent.com/90803397/156860348-72eb598f-60e3-4ace-964f-e9c371981391.png)

That could work, but would require another step to flatten the data out again and then I think we would not see a big win there.

Reply:
I am not sure about your criticism. If you do a groupBy in two keys you get the collections for each of the unique combination. What is the extra step to flatten the data?

Reply:
Okay, maybe I'm missing something. I'll take another look.

Reply:
I've done another revision. Do you mind taking another look?